plugins/source/okta/client/spec_test.go

```diff
@@ -0,0 +1,86 @@
+package client
+
+import (
+	"testing"
+
+	"github.com/cloudquery/codegen/jsonschema"
+)
+
+func TestJSONSchema(t *testing.T) {
+	jsonschema.TestJSONSchema(t, JSONSchema, []jsonschema.TestCase{
+		{
+			Name: "empty spec",
+			Spec: `{}`,
+			Err:  true,
+		},
+		{
+			Name: "spec with token",
+			Spec: `{"token": "tok"}`,
+			Err:  true,
+		},
+		{
+			Name: "spec with domain",
+			Spec: `{"domain": "https://domain.okta.com"}`,
+		},
+		{
+			Name: "spec with token and domain",
+			Spec: `{"token": "tok", "domain": "https://domain.okta.com"}`,
+		},
+		{
+			Name: "spec with token and invalid domain",
+			Spec: `{"token": "tok", "domain": "https://<CHANGE_THIS_TO_YOUR_OKTA_DOMAIN>.okta.com"}`,
+			Err:  true,
+		},
+		{
+			Name: "spec with token and domain and empty rate limit",
+			Spec: `{"token": "tok", "domain": "https://domain.okta.com", "rate_limit": {}}`,
+		},
+		{
+			Name: "spec with token and domain and null rate limit",
+			Spec: `{"token": "tok", "domain": "https://domain.okta.com", "rate_limit": null}`,
+		},
+		{
+			Name: "spec with token and domain and rate limit",
+			Spec: `{"token": "tok", "domain": "https://domain.okta.com", "rate_limit": {"max_backoff": "60s"}}`,
+		},
+		{
+			Name: "spec with token and domain and invalid rate limit",
+			Spec: `{"token": "tok", "domain": "https://domain.okta.com", "rate_limit": {"max_backoff": true}}`,
+			Err:  true,
+		},
+		{
+			Name: "spec with token and domain and zero rate limit",
+			Spec: `{"token": "tok", "domain": "https://domain.okta.com", "rate_limit": {"max_backoff": 0}}`,
```

Comment:
rl tests could be extracted into a separate func, see https://github.com/cloudquery/cloudquery/blob/main/plugins/source/aws/client/spec/organization_test.go

Reply:
done in https://github.com/cloudquery/cloudquery/pull/16493/commits/e8180cc2682a690d964c3790ea1b4d9bcb25b7d8